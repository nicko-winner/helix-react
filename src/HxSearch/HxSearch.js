
import React from 'react';
import PropTypes from 'prop-types';

/**
 * <HxMenu /> wrapper for <hx-search />
 *
 * @example docs https://rackerlabs.github.io/helix-ui/components/search/
 * @example element https://rackerlabs.github.io/helix-ui/elements/hx-search/
 *
 */

class HxSearch extends React.Component {
  componentDidMount() {
    this.hxRef.addEventListener('clear', this.props.onClear);
    this.hxRef.addEventListener('keyup', this.props.onKeyup);
  }

  componentWillUnmount() {
    this.hxRef.removeEventListener('clear', this.props.onClear);
    this.hxRef.removeEventListener('keyup', this.props.onKeyup);
  }

  setRef = (element) => { this.hxRef = element; }
  onBlur = (e) => this.props.onBlur(e);
  onFocus = (e) => this.props.onFocus(e);

  render() {
    const {onClear, onKeyup, onBlur, onFocus, ...hxProps} = this.props;
    return (
      <hx-search
        {...hxProps}
        onFocus={this.onFocus}
        onBlur={this.onBlur}
        ref={this.setRef} />
    );
  }
}

HxSearch.propTypes = {
  onClear: PropTypes.func,
  onKeyup: PropTypes.func,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func,
  value: PropTypes.string,
  disabled: PropTypes.bool,
  placeholder: PropTypes.string
};

HxSearch.defaultPropTypes = {
  onClear: () => null,
  onKeyup: () => null,
  onBlur: () => null,
  onFocus: () => null
};

export default HxSearch;
