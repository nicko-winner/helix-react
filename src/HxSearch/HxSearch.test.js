import * as enzyme from 'enzyme';
import HxSearch from './HxSearch';
import React from 'react';

const mount = (props = {}) => {
  return enzyme.mount(
    <HxSearch {...props} />);
};

describe('HxSearch', () => {
  it('renders correctly', () => {
    expect(mount().html()).toMatchSnapshot();
  });

  it('props.onBlur is called on `blur` event', () => {
    const onBlur = jest.fn();
    mount({onBlur}).simulate('blur');
    expect(onBlur.mock.calls.length).toBe(1);
  });

  it('props.onFocus is called on `focus` event', () => {
    const onFocus = jest.fn();
    mount({onFocus}).simulate('focus');
    expect(onFocus.mock.calls.length).toBe(1);
  });

  it('componentDidMount adds event listeners', () => {
    const instance = mount().instance();
    instance.hxRef.addEventListener = jest.fn();
    instance.componentDidMount();
    const listenersAdded = instance.hxRef.addEventListener.mock.calls.map((event) => event[0]);
    // Note: `focus` and `blur` work without adding / removing event listeners.
    expect(listenersAdded).toEqual(expect.arrayContaining(['keyup', 'clear']));
  });

  it('componentWillUnmount removes all event listeners', () => {
    const instance = mount().instance();
    instance.hxRef.removeEventListener = jest.fn();
    instance.componentWillUnmount();
    const listenersRemoved = instance.hxRef.removeEventListener.mock.calls.map((event) => event[0]);
    // Note: `focus` and `blur` work without adding / removing event listeners.
    expect(listenersRemoved).toEqual(expect.arrayContaining(['keyup', 'clear']));
  });
});
