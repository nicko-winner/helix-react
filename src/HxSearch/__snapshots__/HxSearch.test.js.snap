// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HxSearch renders correctly 1`] = `"<hx-search></hx-search>"`;
